### PR TITLE
Add support for `Net::HTTP#write_timeout` method (Ruby 2.6.0)

### PR DESCRIPTION
**Pull request by springerigor**
[The method](https://ruby-doc.org/stdlib-2.6/libdoc/net/http/rdoc/Net/HTTP.html#method-i-write_timeout-3D) was introduced in Ruby 2.6.0.

The gem already supports `open_timeout` & `read_timeout` options, so it would be nice to provide support for `write_timeout` as well.

From the official documentation:

> Number of seconds to wait for one block to be written (via one write(2) call). Any number may be used, including Floats for fractional seconds. If the HTTP object cannot write data in this many seconds, it raises a Net::WriteTimeout exception. The default value is 60 seconds. Net::WriteTimeout is not raised on Windows.

### CODE REVIEW

**Comment by jnunemaker** [comment]
Seems good to me. @TheSmartnik any thoughts either way?

File: lib/httparty.rb [hunk 1]
```diff
@@ -200,6 +200,19 @@ def read_timeout(t)
...
 
+    # Allows setting a default write_timeout for all HTTP calls in seconds
+    # Supported by Ruby > 2.6.0
+    #
+    #   class Foo
+    #     include HTTParty
+    #     write_timeout 10
+    #   end
+    def write_timeout(t)
+      raise ArgumentError, 'write_timeout must be an integer or float' unless t && (t.is_a?(Integer) || t.is_a?(Float))
```
Comment (TheSmartnik) [inline lib/httparty.rb]: Could extract it to method and refactor other timeout methods as well, please?
Something like `validate_timeout_argument(timeout)` or `check_timeout_argument(timeout)`

File: lib/httparty/connection_adapter.rb [hunk 2]
```diff
@@ -126,6 +133,14 @@ def connection
         http.open_timeout = options[:open_timeout]
       end
 
+      if options[:write_timeout] && (options[:write_timeout].is_a?(Integer) || options[:write_timeout].is_a?(Float))
```
Comment (TheSmartnik) [inline lib/httparty/connection_adapter.rb]: There is a lot of duplication here. Would be better to extract it to a private method `add_timeout?(options[:write_timeout])`

File: lib/httparty/connection_adapter.rb [hunk 3]
```diff
@@ -116,6 +117,12 @@ def connection
       if options[:timeout] && (options[:timeout].is_a?(Integer) || options[:timeout].is_a?(Float))
         http.open_timeout = options[:timeout]
         http.read_timeout = options[:timeout]
+
+        if RUBY_VERSION >= "2.6.0"
+          http.write_timeout = options[:timeout]
+        else
+          Kernel.warn("Warning: option :write_timeout requires Ruby version 2.6.0 or later")
```
Comment (TheSmartnik) [inline lib/httparty/connection_adapter.rb]: It's a timeout for everything, we should set `write_timeout` if we can, but there is no need to add warning here.

File: lib/httparty/connection_adapter.rb [hunk 4]
```diff
@@ -126,6 +133,14 @@ def connection
         http.open_timeout = options[:open_timeout]
       end
 
+      if options[:write_timeout] && (options[:write_timeout].is_a?(Integer) || options[:write_timeout].is_a?(Float))
+        if RUBY_VERSION >= "2.6.0"
```
Comment (TheSmartnik) [inline lib/httparty/connection_adapter.rb]: There is a lot of duplication as well. Probably a would be better, to create a wrapper method like so _(just an idea, not sure about the naming)_
```ruby
for_versions_above('2.6', option: :write_timeout) do
  http.write_timeout = options[:write_timeout]
end
```

File: spec/httparty/connection_adapter_spec.rb [hunk 5]
```diff
@@ -191,6 +200,19 @@
           allow(Net::HTTP).to receive_messages(new: http)
           adapter.connection
         end
+
+        it "should not set the write_timeout" do
+          http = double(
+            "http",
+            :null_object => true,
```
Comment (TheSmartnik) [inline spec/httparty/connection_adapter_spec.rb]: Please, use colon notation for symbol keyword arguments

Comment (springerigor) [reply spec/httparty/connection_adapter_spec.rb]: Hey @TheSmartnik thanks for the comments 👍 I addressed all of them, but this one. I would like to discuss one thing here. 

Due to the fact that the doubles contain names with special characters (e.g. `:use_ssl= => false`) converting them to Ruby 1.9 hash syntax would require using `'` to wrap them. Not sure if it is worth doing in this case. I am totally fine with leaving the current syntax.

Besides that, feel free to take a look at the pushed commits. The code looks better now IMO.

**Comment by springerigor** [comment]
@TheSmartnik A kind reminder for taking another look 👋

**Comment by TheSmartnik** [comment]
@springerigor thanks a lot! Sorry for a long reply

**Pull request merged**